fix: Crash on startup with no recent files

namlit opened the pull request:
- Add nullptr check to getMostRecent method
- Add nullptr check to on_startup function
- Change method descriptions to make explicit that nullptrs can be returned

Fixes #3734

bhennion commented:
This looks great! Thanks for the catch and the fix!
Would you mind rebasing this onto the branch `release-1.1` so that it'd be included in our next minor release?

namlit commented:
> This looks great! Thanks for the catch and the fix! Would you mind rebasing this onto the branch `release-1.1` so that it'd be included in our next minor release?

Just rebased and changed the destination branch for the merge.

bhennion commented:
Thanks!

@xournalpp/core Merging in 48 hours unless someone objects.
